pkg/controllers/user/cis/utils.go

```diff
@@ -107,3 +110,61 @@ func ValidateClusterBeforeLaunchingScan(cluster *v3.Cluster) error {
 	}
 	return nil
 }
+
+func GetBenchmarkVersionToUse(overrideBenchmarkVersion string, currentK8sVersion string,
+	cisConfigLister v3.CisConfigLister, cisConfigClient v3.CisConfigInterface,
+	cisBenchmarkVersionLister v3.CisBenchmarkVersionLister, cisBenchmarkVersionClient v3.CisBenchmarkVersionInterface,
+) (string, bool, error) {
+	bv := overrideBenchmarkVersion
+	shortK8sVersion := util.GetTagMajorVersion(currentK8sVersion)
+	if bv == "" {
+		cisConfigParams, err := kontainerdrivermetadata.GetCisConfigParams(
+			shortK8sVersion,
+			cisConfigLister,
+			cisConfigClient,
+		)
+		if err != nil {
+			logrus.Debugf("cisScanHandler: benchmark version not found for k8s version: %v, using default",
+				shortK8sVersion)
+			cisConfigParams, err = kontainerdrivermetadata.GetCisConfigParams(
+				"default",
+				cisConfigLister,
+				cisConfigClient,
+			)
+			if err != nil {
+				return "", false, fmt.Errorf("cisScanHandler: error fetching default cis config: %v", err)
+			}
+		}
+		bv = cisConfigParams.BenchmarkVersion
+	}
+	benchmarkInfo, err := kontainerdrivermetadata.GetCisBenchmarkVersionInfo(
+		bv,
+		cisBenchmarkVersionLister,
+		cisBenchmarkVersionClient,
+	)
+	if err != nil {
+		return "", false, fmt.Errorf("cisScanHandler: error fetching benchmark version info %v: %v",
+			bv, err)
+	}
+	benchmarkK8sVersionSemVer, err := util.StrToSemVer(ConvertToSemVerStr(benchmarkInfo.MinKubernetesVersion))
+	if err != nil {
+		return "", false, fmt.Errorf("cisScanHandler: error getting sem version for benchmark k8s version %v: %v",
+			benchmarkInfo.MinKubernetesVersion, err)
+	}
+	clusterK8sSemVerStr := ConvertToSemVerStr(shortK8sVersion)
```

Comment:
What if the benchmarkInfo is derived using "default" when kontainerdrivermetadata.GetCisConfigParams("shortK8sVersion"...) has failed? In that case  should we compare with "default"

Reply:
Only CisConfig has "default" which would return a benchmark version (along with min k8s version). There is no "default" in benchmark. So I think we are good here. Let me know if you still have concerns.